source/authentication/custom-jwt.txt

```diff
@@ -9,56 +9,87 @@ Custom JWT Authentication
 .. contents:: On this page
    :local:
    :backlinks: none
-   :depth: 2
+   :depth: 3
    :class: singlecol
 
 .. _custom-jwt-authentication-overview:
 
-Overview
---------
-
 The Custom JWT authentication provider allows users to log in with an
-authentication credential from a third-party system external to Atlas
-App Services. The external system must return a signed :jwt-io:`JSON Web
+authentication credential from a third-party system (external to Atlas
+App Services) and use that token to access data and services with App Services. 
+The external system must return a signed :jwt-io:`JSON Web
 Token (JWT) <introduction>` that contains a unique ID value for the
 authenticated user.
 
-App Services uses the JWT to identify your application's users and
-authenticate their requests. It does not impose any restrictions on the
+Authentication and Authorization
+--------------------------------
+
+The third party JWT provider authenticates users and returns a JWT. App Services 
+uses the JWT to identify your application's users and authorize their requests.
+
+App Services is agnostic to the authentication methods used by the third party 
+provider. It does not impose any restrictions on the
 external authentication system's requirements or authentication methods.
-For example, the system could require the user to perform two factor
-authentication, provide specific credentials, or otherwise identify
-themself.
+For example, the system could require the user to perform multi-factor
+authentication (MFA), provide specific credentials, or otherwise identify
+themselves.
+
+.. _json-web-tokens:
+
+How JWT Authentication Works
+----------------------------
+
+There are many resources online that dive into the intricacies of JWT 
+authentication and the token structure. In the context of App Services, the 
+following diagram provides the process flow of a user logging on to a 
+Device Sync app. The steps below the diagram provide details.
 
 .. figure:: /images/custom-auth-diagram.png
    :alt: Diagram of Custom JWT authentication architecture.
    :figwidth: 750px
 
+JWT authentication with App Services follows these general steps:
+
+1. The user logs on to the third party authentication provider by whatever means 
+   the provider requires.
+
+#. If authentication succeeds, the provider returns a JWT to the client app.
+
+#. The client app logs on to the App Services app, providing the JWT credential.
+
+#. App Services parses and decodes the JWT.
+
+#. If you manually provided signing keys in App Services, App Services checks if 
+   the signing key in the JWT matches one of the signing keys you specified. If 
+   so, the user is authenticated. 
+
+#. If you configured App Services to use a JWK (also referred to JWKS) URI, App 
```

Comment:
I'd like to spell out JWK for the first reference, but this gets awkward fast if we're trying to explain that it can _also_ be JWKS.
```suggestion
#. If you configured App Services to use a JSON Web Key URI (JWK or JWKS URI), App 
```

Reply:
Looks like we use this in the App Services UI content below:
> JSON Web Key Set (JWKS) URI

Should we just standardize on JWKS?

Reply:
Oh, but I see the UI uses "JWK". Ugh. What a mess. My afternoon brain doesn't have any elegant solutions.